vms/platformvm/state/subnet_only_validators.go

```diff
@@ -0,0 +1,383 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/linkeddb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/validators"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/heap"
+	"github.com/ava-labs/avalanchego/vms/components/fee"
+	"github.com/ava-labs/avalanchego/vms/platformvm/block"
+)
+
+var (
+	_ SubnetOnlyValidators = (*subnetOnlyValidators)(nil)
+
+	ErrAlreadyValidator  = errors.New("already a validator")
+	ErrValidatorNotFound = errors.New("validator not found")
+	ErrNonZeroWeight     = errors.New("weight must be 0 when minNonce is MaxUint64")
+)
+
+type SubnetOnlyValidators interface {
```

Comment:
1. Where are we exposing the "Next staker evicted time"?
2. Have we started looking into how the state diff will be implemented here?